### PR TITLE
Adding support for launching an android client over ADB from the editor

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
This PR contains some additional changes that were related or required:

* Updating the local deployment launcher to use a runtime ip if provided.
* Fix AndroidClient build options.
* Mobile connection screen now skips input if IP is provided through launch argument.

#### Tests
I've build and run for the phone a few times with different settings.

### CODE REVIEW

File: workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs [hunk 1]
```diff
@@ -64,5 +72,12 @@ private void TryConnect()
             workerConnector.ConnectionScreenController = this;
             workerConnector.TryConnect();
         }
+
+        private string GetIPFromExtras()
```
Comment (firtina-improbable) [inline workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs]: `FromExtras` is not particularly descriptive or helpful here, would `get receptionist host ip from command line` or something be better?

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 2]
```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
 {
     public static class CommandLineUtility
     {
+        public static Dictionary<string, string> GetArguments()
+        {
+#if UNITY_ANDROID
+            try
+            {
+                var unityPlayer = new UnityEngine.AndroidJavaClass("com.unity3d.player.UnityPlayer");
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: this needs to be within a `using`

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 3]
```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
...
+                var currentActivity = unityPlayer.GetStatic<UnityEngine.AndroidJavaObject>("currentActivity");
+
+                var intent = currentActivity.Call<UnityEngine.AndroidJavaObject>("getIntent");
+                var hasExtra = intent.Call<bool>("hasExtra", "arguments");
+
+                if (hasExtra)
+                {
+                    var extras = intent.Call<UnityEngine.AndroidJavaObject>("getExtras");
+                    var arguments = extras.Call<string>("getString", "arguments");
+                    UnityEngine.Debug.Log(arguments);
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Debug log here can be removed

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 4]
```diff
@@ -57,6 +87,11 @@ private static bool TryGetConfigValue<T>(Dictionary<string, string> dictionary,
                 var flag = args[i];
                 if (flag.StartsWith("+"))
                 {
+                    if (i + 1 >= args.Count)
+                    {
+                        throw new ArgumentException($"Flag \"{flag}\" requires an argument");
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Can you print all arguments here within the exception? It will help the user debug it

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        [MenuItem("SpatialOS/Mobile/Launch on Device", false)]
+        private static void LaunchMobileClient()
+        {
+            EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing APK", 0.3f);
+
+            // Find apk to install
+            if (!TryGetApkPath("build", out var apkPath))
+            {
+                Debug.LogError("Could not find a build out Android binary to launch.");
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]:
```suggestion
                Debug.LogError("Could not find a built out Android binary to launch.");
```

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                return;
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: Clear progress bar here (ideally within a finally clause of the whole method?)

File: workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs [hunk 7]
```diff
@@ -206,6 +206,12 @@ public static void LaunchLocalDeployment()
             var command = Common.SpatialBinary;
             var commandArgs = "local launch";
 
+            var runtimeIp = GdkToolsConfiguration.GetOrCreateInstance().RuntimeIp;
+            if (!string.IsNullOrEmpty(runtimeIp))
+            {
+                commandArgs = $"{commandArgs} --runtime_ip={runtimeIp}";
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs]: Don't we also need to use the `mobile_launch.json` for this?

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 8]
```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
 {
     public static class CommandLineUtility
     {
+        public static Dictionary<string, string> GetArguments()
+        {
+#if UNITY_ANDROID
```
Comment (gkassabli) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: ifdefs like this in generic utility make me sad - in the long run we are going to have a lot of platform-dependent code where it shouldn't be.
The alternative is reworking our processing which would take some effort, so I'm fine with this for now, but we'll have to tackle that issue eventually.

File: workers/unity/Assets/Playground/Config/BuildConfiguration.asset [hunk 9]
```diff
@@ -35,10 +35,10 @@ MonoBehaviour:
     - {fileID: 102900000, guid: 82ccd0fd96a2ba24f88d0c7d01592f2e, type: 3}
     LocalBuildConfig:
       BuildPlatforms: 32
-      BuildOptions: 0
+      BuildOptions: 1
```
Comment (jessicafalk) [inline workers/unity/Assets/Playground/Config/BuildConfiguration.asset]: what are these options now?

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: don't like to have that inside Core. we tried to move everything mobile related to its own package. Let's have a AdbPackage in the Android part of the mobile package

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+using System.Text;
+using Improbable.Gdk.Tools;
+using UnityEditor;
+using UnityEngine;
+
+namespace Improbable.Gdk.Mobile
+{
+    public class LaunchMenu
+    {
+        [MenuItem("SpatialOS/Mobile/Launch on Device", false)]
```
Comment (gkassabli) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: I would rephrase this to Launch on Android Device or similar - people might get confused and try to use this item to launch on iOS device

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+                "-n", $"{bundleId}/com.unity3d.player.UnityPlayerActivity",
+                "-e", "\"arguments\"", $"\\\"{arguments.ToString()}\\\"");
+
+            EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            EditorUtility.ClearProgressBar();
+        }
+
+        private static bool TryGetApkPath(string rootPath, out string apkPath)
+        {
+            foreach (var file in Directory.GetFiles(rootPath, "*.apk", SearchOption.AllDirectories))
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: does a relative path also work correctly on mac?

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+            EditorUtility.DisplayProgressBar("Launching Mobile Client", "Launching Client", 0.9f);
+
+            // Optional arguments to be passed, same as standalone
+            // Use this to pass through the local ip to connect to
+            var runtimeIp = GdkToolsConfiguration.GetOrCreateInstance().RuntimeIp;
+            var arguments = new StringBuilder();
+            if (!string.IsNullOrEmpty(runtimeIp))
+            {
+                arguments.Append($"+receptionistHost {runtimeIp}");
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: `+receptionistHost` -> use the default name we store in the RuntimeConfig (think that's what the class is called)

File: workers/unity/ProjectSettings/GraphicsSettings.asset [hunk 13]
```diff
@@ -37,6 +37,7 @@ GraphicsSettings:
   - {fileID: 10770, guid: 0000000000000000f000000000000000, type: 0}
   - {fileID: 17000, guid: 0000000000000000f000000000000000, type: 0}
   - {fileID: 16000, guid: 0000000000000000f000000000000000, type: 0}
+  - {fileID: 16002, guid: 0000000000000000f000000000000000, type: 0}
```
Comment (jessicafalk) [inline workers/unity/ProjectSettings/GraphicsSettings.asset]: what is that?

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 10 above)
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: add priority as we did for all the other menu items

File: workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs (hunk 7 above)
Comment (gkassabli) [reply workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs]: Correct me if I'm wrong, but mobile_launch.json is only for internal use - we expect external users to use their own config.
That said, allowing to pick launch configuration would be an awesome addition.

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 5 above)
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: add where it is looking for the binary

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+using UnityEngine;
+
+namespace Improbable.Gdk.Mobile
+{
+    public class LaunchMenu
+    {
+        [MenuItem("SpatialOS/Mobile/Launch on Device", false)]
+        private static void LaunchMobileClient()
+        {
+            EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing APK", 0.3f);
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: probably also need a check whether the user actually has adb installed on their machine

File: workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs]: `GetReceptionistHostFromArguments` ?

Comment (firtina-improbable) [reply workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs]: sgtm

File: workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs (hunk 7 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs]: Oh right, now I remember. This was also one of the downsides of using that: there is no way to define another launch config for mobile if we do that.

Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs]: Lets leave that for later

File: workers/unity/Assets/Playground/Config/BuildConfiguration.asset (hunk 9 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Playground/Config/BuildConfiguration.asset]: Debug mode for local, nothing for cloud

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: This is pretty standard to have? I really don't want to do even more packages just to avoid a standard C# flow

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 11 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: MSDN mentiones no differences. "Relative of absolute paths"

File: workers/unity/ProjectSettings/GraphicsSettings.asset (hunk 13 above)
Comment (zeroZshadow) [reply workers/unity/ProjectSettings/GraphicsSettings.asset]: No clue, I assume default settings for android graphics mode we hadn't seen yet

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (gkassabli) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Sorry, misunderstood the goal of the code on the first read.
I'd still vote for restructuring this code and having dedicated class that takes care of mobile/android argument parsing and have here only call that class under ifdef

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 15]
```diff
@@ -157,7 +158,7 @@ PlayerSettings:
     tvOS: com.improbable.gdk
   buildNumber: {}
   AndroidBundleVersionCode: 1
-  AndroidMinSdkVersion: 16
+  AndroidMinSdkVersion: 21
```
Comment (gkassabli) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: Is this intentional?

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: we already have a mobile package?

File: workers/unity/ProjectSettings/GraphicsSettings.asset (hunk 13 above)
Comment (jessicafalk) [reply workers/unity/ProjectSettings/GraphicsSettings.asset]: revert it, if you don't know what it is?

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: I'm still against this as we don't have any mobile code in core and don't want to introduce it here

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 10 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: we would still need to buttons as you need to know if you wanna start an ios or an android device

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 15 above)
Comment (zeroZshadow) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: I think I updated this to ensure Vulkan runs

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Why not?

Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: But we also already have this CommandLineUtility class for all come that works with the command line and parsing it. I don't want this to great a dependency on the mobile package.

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: I agree with @zeroZshadow here - this is a good place for "parse command line arguments for any platform", and it's pretty much the standard way of doing these kinds of things.

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: As the core should only provide the bare minimum to ensure that you can communicate with SpatialOS. If you want to use mobile, you need the mobile feature module anyway, so let's try to keep all mobile related code in one place instead of spreading it all over the place.

(there is also a case for getting rid of this utility and inlining it in the defaultworkerconnector, but that's a different discussion)

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 11 above)
Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: But the question is what would the path be relative to, when you have the project open in Unity Editor. I think it's "likely the same - the project directory", but not 100% sure...

You can use https://docs.unity3d.com/ScriptReference/Application-dataPath.html and the build folder would be `Path.Combine(Application.dataPath, Path.Combine("..", "build"))` guaranteed for all platforms

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 15 above)
Comment (firtina-improbable) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: I guess it's not necessary any more since you were recommending not to use Vulkan as it's not great yet.

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 11 above)
Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: I mean for `private const string rootApkPath = "build";`

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 8 above)
Comment (gkassabli) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: I'd argue for a middle ground here. On one hand, I don't think this belongs in mobile module, but rather in utility module - if we throw everything used in mobile in one module, we'll end up with a mobile module being a bag of logically unrelated things. On the other hand, this code looks a bit alien here - I would separate it in another method (like ExtractAndroidArguments) and call it here under ifdef. Potentially in the future we might want to create a separate class or dedicated mobile utility module

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: while this sounds better, I don't think we should create a new module for one script. The mobile module is already very sparse and contains mainly utility functionality to make it easier, so I don't see a reason why we shouldn't have this method in there as well

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: splitting functionality into different focused module is good, but not if the module is still really small and there is no added benefit in doing it so. We can think about refactoring things out when the module becomes too big

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 10 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: What menu priority should this have? The priorities list for the "other menu items" is current in the tools package, and it is internal.

File: workers/unity/Packages/com.improbable.gdk.mobile/Android/Utility/LaunchArguments.cs [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+                        {
+                            var arguments = extras.Call<string>("getString", "arguments").Split(' ');
+                            return CommandLineUtility.ParseCommandLineArgs(arguments);
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogException(e);
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Android/Utility/LaunchArguments.cs]: when could an exception be thrown? do we just want to log it or actually throw it?

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+                // Find apk to install
+                var rootPath = Path.Combine(Application.dataPath, Path.Combine("..", rootApkPath));
+                if (!TryGetApkPath(rootPath, out var apkPath))
+                {
+                    Debug.LogError($"Could not find a built out Android binary in \"{rootPath}\" to launch.");
+                    return;
+                }
+
+                // Install apk on connected phone / emulator
+                RedirectedProcess.Run("adb", "install", "-r", apkPath);
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: still no check if adb actually exists.

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        [MenuItem(MenuLaunchAndroid, true)]
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: priority missing

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+
+        [MenuItem(MenuLaunchAndroid, false, 73)]
+        private static void LaunchMobileClient()
+        {
+            try
+            {
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing APK", 0.3f);
+
+                // Find apk to install
+                var rootPath = Path.Combine(Application.dataPath, Path.Combine("..", rootApkPath));
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: that could be factored out into a static string, right? we need it for both menu items

File: workers/unity/ProjectSettings/GraphicsSettings.asset (hunk 13 above)
Comment (jessicafalk) [reply workers/unity/ProjectSettings/GraphicsSettings.asset]: still didn't do it

File: workers/unity/Packages/com.improbable.gdk.mobile/Android/Utility/LaunchArguments.cs (hunk 16 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Android/Utility/LaunchArguments.cs]: It can be thrown is this is run on an android version that has breaking changes on this api. Or a newer version of unity that breaks here.
Instead of breaking the entire launch button, it just breaks the auto connect.

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 19 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: It can't, `Application.dataPath` can only be set on runtime.
I could make it a static property however?

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 17 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: Correct, rather out of scope of this initial implementation

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 18 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: The validation function needs no priority

File: workers/unity/ProjectSettings/GraphicsSettings.asset (hunk 13 above)
Comment (zeroZshadow) [reply workers/unity/ProjectSettings/GraphicsSettings.asset]: Its the list of shaders that should always be included. Unity added this when I switched to android build. So I assume it wants it included?

Comment (jessicafalk) [reply workers/unity/ProjectSettings/GraphicsSettings.asset]: never seen that before, so maybe?

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 10 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: yeah, some are in tools, some are in build system. Basically wherever you want it to show up

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 17 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: so what happens right now, if someone hasn't adb installed? cause this will troll ltierally everyone that sees that button and just wants to quickly check out android

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 18 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: we have priorities for all menu items

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 19 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: sure

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 17 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: plus I'm not sure if that will work on mac. I know that we had to use `osascript` for it to work

Comment (gkassabli) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: Yeah, this will most likely fail on Mac. Should we just show error that adb has to be installed on mac? Installing it here might be too convoluted. https://stackoverflow.com/questions/17901692/set-up-adb-on-mac-os-x describes common install ways - none of them look simple enough, though we might download and install it if we really want to

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 20]
```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
...
+                    return ParseCommandLineArgs(arguments.Split(' '));
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogException(e);
+            }
+
+            return new Dictionary<string, string>();
+#else
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Ideally pull this out into a method to have the `#ifdef` around the least code possible. Otherwise it becomes hard to see at a glance what this code can run on. As the other parse logic is already pulled out into a method it would also match that.

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs [hunk 21]
```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
...
+                    UnityEngine.Debug.Log(arguments);
+                    return ParseCommandLineArgs(arguments.Split(' '));
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogException(e);
+            }
+
+            return new Dictionary<string, string>();
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Ideally pull this out into a method to have the #ifdef around the least code possible. Otherwise it becomes hard to see at a glance what this code can run on. As the other parse logic is already pulled out into a method it would also match that.

File: workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs (hunk 18 above)
Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: This is not a menu item though

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: ok, nvm that. just looked up the documentation

Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs]: Its not a menu item

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 21 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: This is what it was, but that was declined by @jessicafalk

**Comment by zeroZshadow** [comment]
adb is no longer required to be on PATH. Now checks to make sure the Android SDK is present

File: workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs (hunk 21 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs]: Actually, what is this even about? This is an outdated code snipped that doesn't even exist anymore.

**Comment by zeroZshadow** [comment]
It now throw an error if: no APK was build out / No AndroidSDK was installed (thus no adb)
It also uses the full path for adb, so it should no longer break if platform-tools isn't on path as it should be.
